src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,244 @@
+  private _tickInterval?: undefined | number;
+
+  public setConfig(config: ClockCardConfig): void {
+    this._config = config;
+    this._initDate();
+  }
+
+  private _initDate() {
+    let locale = this.hass?.locale;
+
+    if (!locale || !this.hass) return;
+
+    if (this._config?.time_format) {
+      locale = { ...locale, time_format: this._config.time_format };
+    }
+
+    this._date = new Intl.DateTimeFormat(this.hass?.locale?.language, {
+      hour: "2-digit",
+      minute: "2-digit",
+      second: "2-digit",
+      hourCycle: useAmPm(locale) ? "h12" : "h23",
+      timeZone: resolveTimeZone(locale.time_zone, this.hass?.config?.time_zone),
+    });
+  }
+
+  public getCardSize(): number {
+    if (this._config?.clock_size === "small") return 1;
+    return 2;
+  }
+
+  public getGridOptions(): LovelaceGridOptions {
+    if (this._config?.clock_size === "small") {
+      return {
+        min_rows: 1,
+        rows: 1,
+        max_rows: 4,
+        min_columns: 4,
+        columns: 6,
+      };
+    }
+
+    if (this._config?.clock_size === "large") {
+      return {
+        min_rows: 2,
+        rows: 2,
+        max_rows: 4,
+        min_columns: 6,
+        columns: 6,
+      };
+    }
+
+    return {
+      min_rows: 1,
+      rows: 2,
+      max_rows: 4,
+      min_columns: 4,
+      columns: 6,
+    };
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._initDate();
+    this._startTick();
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._stopTick();
+  }
+
+  private _startTick() {
+    this._tickInterval = window.setInterval(() => this._tick(), INTERVAL);
+    this._tick();
+  }
+
+  private _stopTick() {
+    if (this._tickInterval) {
+      clearInterval(this._tickInterval);
+      this._tickInterval = undefined;
+    }
+  }
+
+  private _tick() {
+    if (!this._date) return;
+
+    const parts = this._date.formatToParts();
+
+    this._timeHour = parts.find((part) => part.type === "hour")?.value;
+    this._timeMinute = parts.find((part) => part.type === "minute")?.value;
+    this._timeSecond = this._config?.show_seconds
+      ? parts.find((part) => part.type === "second")?.value
+      : undefined;
+    this._timeAmPm = parts.find((part) => part.type === "dayPeriod")?.value;
+  }
+
+  protected render() {
+    if (!this._config) return nothing;
+
+    return html`
+      <ha-card>
+        <div
+          class="time-wrapper ${this._config.clock_size
+            ? `size-${this._config.clock_size}`
+            : ""}"
+        >
+          <div class="time-parts">
+            <div class="time-part hour">${this._timeHour}</div>
+            <div class="time-part minute">${this._timeMinute}</div>
+            <div class="time-side">
+              ${this._timeSecond !== undefined
+                ? html`<div class="time-part second">${this._timeSecond}</div>`
+                : nothing}
+              ${this._config.time_format === TimeFormat.am_pm
```

Comment:
If the default of the user is am/pm it will not be shown now?
```suggestion
              ${this._timeAmPm
```